pkg/reconciler/openshift/openshift.go

```diff
@@ -0,0 +1,25 @@
+package openshift
+
+import (
+	argoprojv1alpha1 "github.com/argoproj-labs/argocd-operator/pkg/apis/argoproj/v1alpha1"
+	v1 "k8s.io/api/rbac/v1"
+
+	"github.com/argoproj-labs/argocd-operator/pkg/controller/argocd"
+)
+
+func init() {
+	argocd.Register("reconciler", reconcilerHook)
+}
+
+func reconcilerHook(cr *argoprojv1alpha1.ArgoCD, v interface{}) error {
```

Comment:
There are no tests for the reconcilerHook?

We should isolate all the openshift specific code to this package, starting with the GVK extensions.